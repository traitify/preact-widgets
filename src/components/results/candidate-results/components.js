export CandidateResults from "./index";
export PersonalityArchetype from "./personality-archetype";
export PersonalityDetails from "./personality-details";
export PersonalityDimensionColumn from "./personality-dimension-column";
export PersonalityDimensionColumns from "./personality-dimension-columns";
export PersonalityDimensionDetails from "./personality-dimension-details";
export PersonalityDimensions from "./personality-dimensions";
