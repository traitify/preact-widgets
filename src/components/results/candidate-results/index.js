import {
  faThLarge,
  faUser
} from "@fortawesome/free-solid-svg-icons";
import PropTypes from "prop-types";
import Icon from "lib/helpers/icon";
import withTraitify from "lib/with-traitify";
import Guide from "components/results/guide";
import PersonalityArchetype from "components/results/personality/archetype/heading";
import PersonalityDimensions from "components/results/personality/dimension/list";
import PersonalityTips from "components/results/personality/archetype/tips";
import PersonalityTraits from "components/results/personality/trait/list";
import style from "../style.scss";

function CandidateResults(props) {
  const {getOption, translate} = props;
  let disabledComponents = getOption("disabledComponents") || [];
  const allowHeaders = getOption("allowHeaders");
  const thirdPerson = getOption("perspective") === "thirdPerson";
  if(!thirdPerson) { disabledComponents = [...disabledComponents, "PersonalityPitfalls"]; }

  return (
    <section>
      {allowHeaders && (
        <p className={style.heading}><Icon icon={faUser} /> {translate("personality_type")}</p>
      )}
      <PersonalityArchetype {...props} />
      {allowHeaders && (
        <p className={style.heading}><Icon icon={faThLarge} /> {translate("personality_details")}</p>
      )}
      <PersonalityTips {...props} />
      {allowHeaders && (
        <p className={style.lessMarginBottom}><Icon icon={faThLarge} /> {translate("personality_advice")}</p>
      )}
      <PersonalityDimensions {...props} disabledComponents={disabledComponents} />
      {thirdPerson && (
        <div>
          {allowHeaders && (
            <p className={style.heading}><Icon icon={faThLarge} /> {translate("interview_guide_heading")}</p>
          )}
          <Guide {...props} />
        </div>
      )}
      {allowHeaders && (
        <p className={style.paddingTop}><Icon icon={faThLarge} /> {translate("personality_traits")}</p>
      )}
      <PersonalityTraits {...props} />
    </section>
  );
}

CandidateResults.propTypes = {
  getOption: PropTypes.func.isRequired,
  translate: PropTypes.func.isRequired
};

export default withTraitify(CandidateResults);
