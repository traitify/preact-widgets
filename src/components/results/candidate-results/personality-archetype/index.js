import PropTypes from "prop-types";
import {Component} from "react";
import DangerousHTML from "lib/helpers/dangerous-html";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import style from "./style.scss";

class PersonalityArchetype extends Component {
  static defaultProps = {assessment: null, deck: null}
  static propTypes = {
    assessment: PropTypes.shape({archetype: PropTypes.object}),
    deck: PropTypes.shape({name: PropTypes.string.isRequired}),
    followDeck: PropTypes.func.isRequired,
    getOption: PropTypes.func.isRequired,
    isReady: PropTypes.func.isRequired,
    options: PropTypes.shape({archetype: PropTypes.object}).isRequired,
    translate: PropTypes.func.isRequired,
    ui: TraitifyPropTypes.ui.isRequired
  }
  componentDidMount() {
    this.props.ui.trigger("PersonalityArchetype.initialized", this);
    this.props.followDeck();
  }
  componentDidUpdate() {
    this.props.ui.trigger("PersonalityArchetype.updated", this);
  }
  render() {
    if(!this.props.isReady("results")) { return null; }
    if(!this.props.isReady("deck")) { return null; }

    const disabledComponents = this.props.getOption("disabledComponents") || [];
    if(disabledComponents.includes("PersonalityArchetype")) { return null; }

    const personality = this.props.assessment.archetype;
    if(!personality) { return null; }
    const badge = personality.details.find(({title}) => title === "Badge");
    const perspective = this.props.getOption("perspective");
    const description = (
      perspective === "thirdPerson"
        ? personality.details.find(({title}) => title === "Hiring Manager Description")
        : personality.details.find(({title}) => title === "Candidate Description")
    );
    const splitIfPresent = description && style.badgeAndName;

    return (
      <div className={style.container}>
        {perspective === "thirdPerson" ? (
          <div className={style.thirdPersonDetails}>
            <div className={splitIfPresent}>
              {badge && <img alt={personality.name} src={badge.body} />}
              <DangerousHTML
                html={this.props.translate("personality_heading_third_person", {
                  deck_name: this.props.deck.name,
                  personality: `<br /><span>${personality.name}</span><br />`
                })}
                className={style.personalityHeading}
                tag="h2"
              />
            </div>
            {description && <span className={style.divider} />}
            {description && <span className={style.body}>{description.body}</span>}
          </div>
        ) : [
          <div key="heading" className={style.details}>
            {badge && <img alt={personality.name} src={badge.body} />}
            <DangerousHTML
              html={this.props.translate("personality_heading", {
                deck_name: this.props.deck.name,
                personality: `<span>${personality.name}</span>`
              })}
              tag="h2"
            />
            {description && <p>{description.body}</p>}
          </div>,
          <div key="meaning" className={style.meaning}>
            <DangerousHTML html={this.props.translate("candidate_description_for_archetype_html")} tag="p" />
          </div>
        ]}
      </div>
    );
  }
}

export {PersonalityArchetype as Component};
export default withTraitify(PersonalityArchetype);
