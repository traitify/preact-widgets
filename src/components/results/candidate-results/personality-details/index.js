import PropTypes from "prop-types";
import {Component} from "react";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import style from "./style";

class PersonalityDetails extends Component {
  static defaultProps = {assessment: null}
  static propTypes = {
    assessment: PropTypes.shape({archetype: PropTypes.object}),
    getOption: PropTypes.func.isRequired,
    isReady: PropTypes.func.isRequired,
    translate: PropTypes.func.isRequired,
    ui: TraitifyPropTypes.ui.isRequired
  }
  componentDidMount() {
    this.props.ui.trigger("PersonalityDetails.initialized", this);
  }
  componentDidUpdate() {
    this.props.ui.trigger("PersonalityDetails.updated", this);
  }
  render() {
    if(!this.props.isReady("results")) { return null; }

    const personality = this.props.assessment.archetype || {};
    const {details} = personality;
    if(!details) { return null; }

    const disabledComponents = this.props.getOption("disabledComponents") || [];
    let disableSettings = disabledComponents.includes("PersonalitySettings");
    let disableTools = disabledComponents.includes("PersonalityTools");
    const settings = details.filter(({title}) => (title === "Settings that Work for You")).map(({body}) => body);
    const tools = details.filter(({title}) => (title === "Tools to Use")).map(({body}) => body);
    if(settings.length === 0) { disableSettings = true; }
    if(tools.length === 0) { disableTools = true; }
    if(disableSettings && disableTools) { return null; }

    return (
      <div className={style.container}>
        {!disableTools && (
          <div className={style.detail}>
            <div className={style.content}>
              <div className={style.bar} style={{background: "#008dc7"}} />
              <h4 className={style.title} style={{color: "#008dc7"}}>{this.props.translate("candidate_heading_for_tools")}</h4>
              <ul className={style.description}>
                {tools.map((tool) => (
                  <li key={tool}>{tool}</li>
                ))}
              </ul>
            </div>
          </div>
        )}
        {!disableSettings && (
          <div className={style.detail}>
            <div className={style.content}>
              <div className={style.bar} style={{background: "#32be4b"}} />
              <h4 className={style.title} style={{color: "#32be4b"}}>{this.props.translate("candidate_heading_for_settings")}</h4>
              <ul className={style.description}>
                {settings.map((setting) => (
                  <li key={setting}>{setting}</li>
                ))}
              </ul>
            </div>
          </div>
        )}
        <div className={style.clearfix} />
      </div>
    );
  }
}

export {PersonalityDetails as Component};
export default withTraitify(PersonalityDetails);
