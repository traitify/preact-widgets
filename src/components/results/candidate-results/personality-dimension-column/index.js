import PropTypes from "prop-types";
import {Component} from "react";
import {rgba} from "lib/helpers/color";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import style from "./style.scss";

class PersonalityDimensionColumn extends Component {
  static propTypes = {
    translate: PropTypes.func.isRequired,
    type: PropTypes.shape({
      personality_type: PropTypes.shape({
        badge: PropTypes.shape({
          color_1: PropTypes.string.isRequired,
          image_medium: PropTypes.string.isRequired
        }).isRequired,
        level: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired
      }).isRequired,
      score: PropTypes.number.isRequired
    }).isRequired,
    ui: TraitifyPropTypes.ui.isRequired
  }
  componentDidMount() {
    this.props.ui.trigger("PersonalityDimensionColumn.initialized", this);
  }
  componentDidUpdate() {
    this.props.ui.trigger("PersonalityDimensionColumn.updated", this);
  }
  render() {
    const {translate, type: {personality_type: {badge, level, name}, score}} = this.props;
    const color = `#${badge.color_1}`;
    let backgroundClass;
    switch(true) {
      case score <= 3: backgroundClass = "low"; break;
      case score <= 6: backgroundClass = "medium"; break;
      default: backgroundClass = "high";
    }

    return (
      <li className={style.container}>
        <div>
          <div className={style.icon} style={{border: `3px solid ${color}`, background: rgba(color, 8.5)}}>
            <img src={badge.image_medium} alt={`${name} ${translate("badge")}`} />
          </div>
          <h3>{name}</h3>
          <h2>{level}</h2>
        </div>
        <div className={`${style.background} ${style[backgroundClass]}`} style={{background: rgba(color, 10)}} />
      </li>
    );
  }
}

export {PersonalityDimensionColumn as Component};
export default withTraitify(PersonalityDimensionColumn);
