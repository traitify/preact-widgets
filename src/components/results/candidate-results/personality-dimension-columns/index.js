import PropTypes from "prop-types";
import {Component} from "react";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import PersonalityDimensionColumn from "../personality-dimension-column";
import style from "./style";

class PersonalityDimensionColumns extends Component {
  static defaultProps = {assessment: null}
  static propTypes = {
    assessment: PropTypes.shape({personality_types: PropTypes.array}),
    isReady: PropTypes.func.isRequired,
    translate: PropTypes.func.isRequired,
    ui: TraitifyPropTypes.ui.isRequired
  }
  componentDidMount() {
    this.props.ui.trigger("PersonalityDimensionColumns.initialized", this);
  }
  componentDidUpdate() {
    this.props.ui.trigger("PersonalityDimensionColumns.updated", this);
  }
  render() {
    if(!this.props.isReady("results")) { return null; }

    const types = this.props.assessment.personality_types.sort((x, y) => {
      const xDetail = x.personality_type.details.find(({title}) => title === "Position") || {};
      const yDetail = y.personality_type.details.find(({title}) => title === "Position") || {};

      return (xDetail.body || 1) - (yDetail.body || 1);
    });

    return (
      <div className={style.container}>
        <p>{this.props.translate("candidate_description_for_dimensions")}</p>
        <ul className={style.columns}>
          {types.map((type) => (
            <PersonalityDimensionColumn
              key={type.personality_type.id}
              type={type}
              {...this.props}
            />
          ))}
        </ul>
        <div className={style.columnsBuffer} />
      </div>
    );
  }
}

export {PersonalityDimensionColumns as Component};
export default withTraitify(PersonalityDimensionColumns);
