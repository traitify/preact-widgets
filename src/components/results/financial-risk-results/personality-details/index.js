import PropTypes from "prop-types";
import {Component} from "react";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import style from "./style";

class PersonalityDetails extends Component {
  static defaultProps = {assessment: null}
  static propTypes = {
    assessment: PropTypes.shape({archetype: PropTypes.object}),
    isReady: PropTypes.func.isRequired,
    ui: TraitifyPropTypes.ui.isRequired
  }
  componentDidMount() {
    this.props.ui.trigger("PersonalityDetails.initialized", this);
  }
  componentDidUpdate() {
    this.props.ui.trigger("PersonalityDetails.updated", this);
  }
  render() {
    if(!this.props.isReady("results")) { return null; }

    // const personality = this.props.assessment.archetype || {};

    return (
      <div className={style.details}>
        <div className={style.detailsLife}>
          <h2>The Neutral Financial Risk Style in everyday life:</h2>
          <p>You are unenthused by a lack of change in life and thrive with unpredictability.</p>
        </div>
        <div className={style.detailsFinancial}>
          <h2>The Neutral Financial Risk Style in financial decisions:</h2>
          <p>You are unenthused by a lack of change in life and thrive with unpredictability.</p>
        </div>
      </div>
    );
  }
}

export {PersonalityDetails as Component};
export default withTraitify(PersonalityDetails);
