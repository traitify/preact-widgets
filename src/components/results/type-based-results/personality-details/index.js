import { h, Component } from "preact";
import style from "./style";

export default class PersonalityDetails extends Component {
  componentDidMount(){
    this.props.triggerCallback("personalitydetails", "initialized", this);
  }
  render() {
    let personality = this.props.assessment.personality_blend;
    personality = personality || this.props.assessment.personality_types[0];
    let details = personality.details;
    let complement = details.find(d => d.title == "Complement");
    complement = complement && complement.body;
    let conflict = details.find(d => d.title == "Conflict");
    conflict = conflict && conflict.body;
    let environments = personality.environments || [];
    return (
      <div class={style.details}>
        {complement && (
          <div class={style.complements}>
            <h4>Complements</h4>
            <p>{complement}</p>
          </div>
        )}
        {conflict && (
          <div class={style.conflicts}>
            <h4>Conflicts</h4>
            <p>{conflict}</p>
          </div>
        )}
        {environments[0] && (
          <div class={style.environment}>
            <h4>Best Work Environment</h4>
            <ul>
              {environments.map(environment => {
                return <li>{environment.name}</li>;
              })}
            </ul>
          </div>
        )}
      </div>
    );
  }
}
