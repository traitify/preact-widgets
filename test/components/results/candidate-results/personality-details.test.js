import {Component} from "components/results/candidate-results/personality-details";
import ComponentHandler from "support/component-handler";
import assessment from "support/json/assessment/dimension-based.json";

jest.mock("lib/helpers", () => ({detailWithPerspective: jest.fn().mockImplementation((options) => options.base[options.name] || options.name)}));
jest.mock("lib/with-traitify", () => ((value) => value));

const cautions = [
  {body: "Be Cautious", title: "Caution Zone"},
  {body: "Of People", title: "Caution Zone"}
];
const settings = [
  {body: "Encourages peer support in pursuit of goals", title: "Settings that Work for You"},
  {body: "De-emphasizes competition among colleagues", title: "Settings that Work for You"},
  {body: "Often has a fast paced and energetic vibe", title: "Settings that Work for You"},
  {body: "Includes opportunities for large group discussions", title: "Settings that Work for You"},
  {body: "Values creativity", title: "Settings that Work for You"},
  {body: "Has opportunities for innovation and exploration", title: "Settings that Work for You"},
  {body: "Fueled by a sense of duty", title: "Settings that Work for You"},
  {body: "Prioritizes work quality", title: "Settings that Work for You"},
  {body: "Allows individuals to deal with their own setbacks", title: "Settings that Work for You"},
  {body: "Requires people to respond to challenges", title: "Settings that Work for You"}
];
const tools = [
  {body: "Compatible with many other personalities", title: "Tools to Use"},
  {body: "Sought after as a good person to dialogue with", title: "Tools to Use"},
  {body: "Work at a pace that suits everyone", title: "Tools to Use"},
  {body: "High energy, outgoing", title: "Tools to Use"},
  {body: "Comfortable in the limelight", title: "Tools to Use"},
  {body: "Have effective coping strategies for work stress", title: "Tools to Use"}
];

const assessmentWith = (details) => ({
  ...assessment, archetype: {...assessment.archetype, details}
});

describe("PersonalityDetails", () => {
  let props;

  beforeEach(() => {
    props = {
      assessment: assessmentWith([...cautions, ...settings, ...tools]),
      getOption: jest.fn().mockName("getOption"),
      isReady: jest.fn().mockName("isReady").mockReturnValue(true),
      translate: jest.fn().mockName("translate").mockImplementation((value, options = {}) => `${value}, ${options}`),
      ui: {
        current: {},
        off: jest.fn().mockName("off"),
        on: jest.fn().mockName("on"),
        trigger: jest.fn().mockName("trigger")
      }
    };
  });

  describe("callbacks", () => {
    it("triggers initialization", () => {
      const component = new ComponentHandler(<Component {...props} />);

      expect(props.ui.trigger).toHaveBeenCalledWith("PersonalityDetails.initialized", component.instance);
    });

    it("triggers update", () => {
      const component = new ComponentHandler(<Component {...props} />);
      component.updateProps();

      expect(props.ui.trigger).toHaveBeenCalledWith("PersonalityDetails.updated", component.instance);
    });
  });

  it("renders component", () => {
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders component with cautions", () => {
    props.getOption.mockImplementation((key) => (key === "perspective" ? "thirdPerson" : []));
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if disabled", () => {
    props.getOption.mockReturnValue(["PersonalitySettings", "PersonalityTools"]);
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if results not ready", () => {
    props.assessment = null;
    props.isReady.mockImplementation((value) => value !== "results");
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if no archetype", () => {
    props.assessment = {...props.assessment, archetype: null};
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders nothing if no details", () => {
    props.assessment = assessmentWith([]);
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders settings if no tools", () => {
    props.assessment = assessmentWith([...settings]);
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders settings if tools disabled", () => {
    props.getOption.mockReturnValue(["PersonalityTools"]);
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders tools if no settings", () => {
    props.assessment = assessmentWith([...tools]);
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });

  it("renders tools if settings disabled", () => {
    props.getOption.mockReturnValue(["PersonalitySettings"]);
    const component = new ComponentHandler(<Component {...props} />);

    expect(component.tree).toMatchSnapshot();
  });
});
