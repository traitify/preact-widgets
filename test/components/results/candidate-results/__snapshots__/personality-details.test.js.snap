// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalityDetails renders component 1`] = `
<div>
  <div>
    <div>
      <div
        style={
          Object {
            "background": "#008dc7",
          }
        }
      />
      <h4
        style={
          Object {
            "color": "#008dc7",
          }
        }
      >
        candidate_heading_for_tools, [object Object]
      </h4>
      <ul>
        <li>
          Compatible with many other personalities
        </li>
        <li>
          Sought after as a good person to dialogue with
        </li>
        <li>
          Work at a pace that suits everyone
        </li>
        <li>
          High energy, outgoing
        </li>
        <li>
          Comfortable in the limelight
        </li>
        <li>
          Have effective coping strategies for work stress
        </li>
      </ul>
    </div>
  </div>
  <div>
    <div>
      <div
        style={
          Object {
            "background": "#32be4b",
          }
        }
      />
      <h4
        style={
          Object {
            "color": "#32be4b",
          }
        }
      >
        candidate_heading_for_settings, [object Object]
      </h4>
      <ul>
        <li>
          Encourages peer support in pursuit of goals
        </li>
        <li>
          De-emphasizes competition among colleagues
        </li>
        <li>
          Often has a fast paced and energetic vibe
        </li>
        <li>
          Includes opportunities for large group discussions
        </li>
        <li>
          Values creativity
        </li>
        <li>
          Has opportunities for innovation and exploration
        </li>
        <li>
          Fueled by a sense of duty
        </li>
        <li>
          Prioritizes work quality
        </li>
        <li>
          Allows individuals to deal with their own setbacks
        </li>
        <li>
          Requires people to respond to challenges
        </li>
      </ul>
    </div>
  </div>
  <div />
</div>
`;

exports[`PersonalityDetails renders component with cautions 1`] = `
<div>
  <div>
    <div>
      <div
        style={
          Object {
            "background": "#008dc7",
          }
        }
      />
      <h4
        style={
          Object {
            "color": "#008dc7",
          }
        }
      >
        candidate_heading_for_tools, [object Object]
      </h4>
      <ul>
        <li>
          Compatible with many other personalities
        </li>
        <li>
          Sought after as a good person to dialogue with
        </li>
        <li>
          Work at a pace that suits everyone
        </li>
        <li>
          High energy, outgoing
        </li>
        <li>
          Comfortable in the limelight
        </li>
        <li>
          Have effective coping strategies for work stress
        </li>
      </ul>
    </div>
  </div>
  <div>
    <div>
      <div
        style={
          Object {
            "background": "#ef615e",
          }
        }
      />
      <h4
        style={
          Object {
            "color": "#ef615e",
          }
        }
      >
        caution_zone, [object Object]
      </h4>
      <ul>
        <li>
          Be Cautious
        </li>
        <li>
          Of People
        </li>
      </ul>
    </div>
  </div>
  <div>
    <div>
      <div
        style={
          Object {
            "background": "#32be4b",
          }
        }
      />
      <h4
        style={
          Object {
            "color": "#32be4b",
          }
        }
      >
        candidate_heading_for_settings, [object Object]
      </h4>
      <ul>
        <li>
          Encourages peer support in pursuit of goals
        </li>
        <li>
          De-emphasizes competition among colleagues
        </li>
        <li>
          Often has a fast paced and energetic vibe
        </li>
        <li>
          Includes opportunities for large group discussions
        </li>
        <li>
          Values creativity
        </li>
        <li>
          Has opportunities for innovation and exploration
        </li>
        <li>
          Fueled by a sense of duty
        </li>
        <li>
          Prioritizes work quality
        </li>
        <li>
          Allows individuals to deal with their own setbacks
        </li>
        <li>
          Requires people to respond to challenges
        </li>
      </ul>
    </div>
  </div>
  <div />
</div>
`;

exports[`PersonalityDetails renders nothing if disabled 1`] = `null`;

exports[`PersonalityDetails renders nothing if no archetype 1`] = `null`;

exports[`PersonalityDetails renders nothing if no details 1`] = `null`;

exports[`PersonalityDetails renders nothing if results not ready 1`] = `null`;

exports[`PersonalityDetails renders settings if no tools 1`] = `
<div>
  <div>
    <div>
      <div
        style={
          Object {
            "background": "#32be4b",
          }
        }
      />
      <h4
        style={
          Object {
            "color": "#32be4b",
          }
        }
      >
        candidate_heading_for_settings, [object Object]
      </h4>
      <ul>
        <li>
          Encourages peer support in pursuit of goals
        </li>
        <li>
          De-emphasizes competition among colleagues
        </li>
        <li>
          Often has a fast paced and energetic vibe
        </li>
        <li>
          Includes opportunities for large group discussions
        </li>
        <li>
          Values creativity
        </li>
        <li>
          Has opportunities for innovation and exploration
        </li>
        <li>
          Fueled by a sense of duty
        </li>
        <li>
          Prioritizes work quality
        </li>
        <li>
          Allows individuals to deal with their own setbacks
        </li>
        <li>
          Requires people to respond to challenges
        </li>
      </ul>
    </div>
  </div>
  <div />
</div>
`;

exports[`PersonalityDetails renders settings if tools disabled 1`] = `
<div>
  <div>
    <div>
      <div
        style={
          Object {
            "background": "#32be4b",
          }
        }
      />
      <h4
        style={
          Object {
            "color": "#32be4b",
          }
        }
      >
        candidate_heading_for_settings, [object Object]
      </h4>
      <ul>
        <li>
          Encourages peer support in pursuit of goals
        </li>
        <li>
          De-emphasizes competition among colleagues
        </li>
        <li>
          Often has a fast paced and energetic vibe
        </li>
        <li>
          Includes opportunities for large group discussions
        </li>
        <li>
          Values creativity
        </li>
        <li>
          Has opportunities for innovation and exploration
        </li>
        <li>
          Fueled by a sense of duty
        </li>
        <li>
          Prioritizes work quality
        </li>
        <li>
          Allows individuals to deal with their own setbacks
        </li>
        <li>
          Requires people to respond to challenges
        </li>
      </ul>
    </div>
  </div>
  <div />
</div>
`;

exports[`PersonalityDetails renders tools if no settings 1`] = `
<div>
  <div>
    <div>
      <div
        style={
          Object {
            "background": "#008dc7",
          }
        }
      />
      <h4
        style={
          Object {
            "color": "#008dc7",
          }
        }
      >
        candidate_heading_for_tools, [object Object]
      </h4>
      <ul>
        <li>
          Compatible with many other personalities
        </li>
        <li>
          Sought after as a good person to dialogue with
        </li>
        <li>
          Work at a pace that suits everyone
        </li>
        <li>
          High energy, outgoing
        </li>
        <li>
          Comfortable in the limelight
        </li>
        <li>
          Have effective coping strategies for work stress
        </li>
      </ul>
    </div>
  </div>
  <div />
</div>
`;

exports[`PersonalityDetails renders tools if settings disabled 1`] = `
<div>
  <div>
    <div>
      <div
        style={
          Object {
            "background": "#008dc7",
          }
        }
      />
      <h4
        style={
          Object {
            "color": "#008dc7",
          }
        }
      >
        candidate_heading_for_tools, [object Object]
      </h4>
      <ul>
        <li>
          Compatible with many other personalities
        </li>
        <li>
          Sought after as a good person to dialogue with
        </li>
        <li>
          Work at a pace that suits everyone
        </li>
        <li>
          High energy, outgoing
        </li>
        <li>
          Comfortable in the limelight
        </li>
        <li>
          Have effective coping strategies for work stress
        </li>
      </ul>
    </div>
  </div>
  <div />
</div>
`;
